assert/assertions.go

```diff
@@ -87,55 +87,96 @@ func ObjectsAreEqualValues(expected, actual interface{}) bool {
 }
 
 /* CallerInfo is necessary because the assert functions use the testing object
-internally, causing it to print the file:line of the assert method, rather than where
-the problem actually occured in calling code.*/
+internally, causing it to print the file:line of the assert method, rather than
+where the problem actually occurred in calling code.*/
 
 // CallerInfo returns an array of strings containing the file and line number
-// of each stack frame leading from the current test to the assert call that
-// failed.
+// of each stack frame leading from the assert call that failed back to the
+// current test entry point.
 func CallerInfo() []string {
+	// 16 is how many traces we want to collect, its the call stack between
+	// CallerInfo and up toward the program entry point. As there is about 7
+	// frames above CallerInfo() until the usercode, this leaves about 8 frames
+	// for the returned stacktrace. The goal is to provide an informative error
+	// msg, not a scrolling game.
+	st := make([]uintptr, 16)
 
-	pc := uintptr(0)
-	file := ""
-	line := 0
-	ok := false
-	name := ""
-
-	callers := []string{}
-	for i := 0; ; i++ {
-		pc, file, line, ok = runtime.Caller(i)
-		if !ok {
-			return nil
+	// And here we already skip 2 stack frames, the Callers function frame and
+	// our own CallerInfo frame.
+	stDepth := runtime.Callers(2, st)
+
+	r := []string{}
```

Comment:
For empty slices, use

    var r []string

https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices